### PR TITLE
Callisto Hardfork Meta

### PR DESCRIPTION
**Pull request by ghost**
Added 

+ Callisto Mainnet support with multi-geth compatiblity
+ Default mainnet will be callisto and default data dir is ~/.callisto which is same with CLO/1.0
+ First draft hardfork support for CLOHF1 is now added, please change the address when the contract is deployed

Further steps needed

+ Change Callisto stake address to point out deployed contract
+ Deploy POW testnet for mainnet implementation
+ Enable [Appveyor CI](https://www.appveyor.com) for automatic windows build

Reward status:

600 CLO per block reward:

- 120 CLO for treasury
- 60 CLO for stake
- 420 CLO for miner

After HF1

600 CLO per block reward:

- 60 CLO for treasury
- 120 CLO for stake
- 420 CLO for miner

### CODE REVIEW

**Comment by ghost** [comment]
Local Test Suite
```
?       github.com/ethereum/go-ethereum [no test files]
ok      github.com/ethereum/go-ethereum/accounts        (cached)
ok      github.com/ethereum/go-ethereum/accounts/abi    (cached)
ok      github.com/ethereum/go-ethereum/accounts/abi/bind       3.786s
?       github.com/ethereum/go-ethereum/accounts/abi/bind/backends      [no test files]
ok      github.com/ethereum/go-ethereum/accounts/keystore       16.151s
?       github.com/ethereum/go-ethereum/accounts/usbwallet      [no test files]
?       github.com/ethereum/go-ethereum/accounts/usbwallet/internal/trezor      [no test files]
?       github.com/ethereum/go-ethereum/cmd/abigen      [no test files]
?       github.com/ethereum/go-ethereum/cmd/bootnode    [no test files]
?       github.com/ethereum/go-ethereum/cmd/clef        [no test files]
ok      github.com/ethereum/go-ethereum/cmd/ethkey      (cached)
?       github.com/ethereum/go-ethereum/cmd/evm [no test files]
?       github.com/ethereum/go-ethereum/cmd/evm/internal/compiler       [no test files]
?       github.com/ethereum/go-ethereum/cmd/faucet      [no test files]
ok      github.com/ethereum/go-ethereum/cmd/geth        (cached)
?       github.com/ethereum/go-ethereum/cmd/internal/browser    [no test files]
?       github.com/ethereum/go-ethereum/cmd/p2psim      [no test files]
?       github.com/ethereum/go-ethereum/cmd/puppeth     [no test files]
?       github.com/ethereum/go-ethereum/cmd/rlpdump     [no test files]
ok      github.com/ethereum/go-ethereum/cmd/swarm       (cached)
?       github.com/ethereum/go-ethereum/cmd/swarm/swarm-smoke   [no test files]
ok      github.com/ethereum/go-ethereum/cmd/utils       (cached)
?       github.com/ethereum/go-ethereum/cmd/wnode       [no test files]
ok      github.com/ethereum/go-ethereum/common  (cached)
ok      github.com/ethereum/go-ethereum/common/bitutil  (cached)
ok      github.com/ethereum/go-ethereum/common/compiler (cached)
ok      github.com/ethereum/go-ethereum/common/fdlimit  (cached)
ok      github.com/ethereum/go-ethereum/common/hexutil  (cached)
ok      github.com/ethereum/go-ethereum/common/math     (cached)
?       github.com/ethereum/go-ethereum/common/mclock   [no test files]
?       github.com/ethereum/go-ethereum/consensus       [no test files]
ok      github.com/ethereum/go-ethereum/consensus/clique        (cached)
ok      github.com/ethereum/go-ethereum/consensus/ethash        (cached)
?       github.com/ethereum/go-ethereum/consensus/misc  [no test files]
ok      github.com/ethereum/go-ethereum/console (cached)
ok      github.com/ethereum/go-ethereum/contracts/chequebook    (cached)
?       github.com/ethereum/go-ethereum/contracts/chequebook/contract   [no test files]
ok      github.com/ethereum/go-ethereum/contracts/ens   (cached)
?       github.com/ethereum/go-ethereum/contracts/ens/contract  [no test files]
ok      github.com/ethereum/go-ethereum/core    (cached)
ok      github.com/ethereum/go-ethereum/core/asm        (cached)
ok      github.com/ethereum/go-ethereum/core/bloombits  (cached)
ok      github.com/ethereum/go-ethereum/core/rawdb      (cached)
ok      github.com/ethereum/go-ethereum/core/state      (cached)
ok      github.com/ethereum/go-ethereum/core/types      (cached)
ok      github.com/ethereum/go-ethereum/core/vm (cached)
ok      github.com/ethereum/go-ethereum/core/vm/runtime (cached)
ok      github.com/ethereum/go-ethereum/crypto  (cached)
?       github.com/ethereum/go-ethereum/crypto/bn256    [no test files]
ok      github.com/ethereum/go-ethereum/crypto/bn256/cloudflare (cached)
ok      github.com/ethereum/go-ethereum/crypto/bn256/google     (cached)
ok      github.com/ethereum/go-ethereum/crypto/ecies    (cached)
ok      github.com/ethereum/go-ethereum/crypto/secp256k1        (cached)
ok      github.com/ethereum/go-ethereum/crypto/sha3     (cached)
?       github.com/ethereum/go-ethereum/dashboard       [no test files]
ok      github.com/ethereum/go-ethereum/eth     (cached)
ok      github.com/ethereum/go-ethereum/eth/downloader  78.227s
ok      github.com/ethereum/go-ethereum/eth/fetcher     10.112s
ok      github.com/ethereum/go-ethereum/eth/filters     2.340s
?       github.com/ethereum/go-ethereum/eth/gasprice    [no test files]
ok      github.com/ethereum/go-ethereum/eth/tracers     0.164s
?       github.com/ethereum/go-ethereum/eth/tracers/internal/tracers    [no test files]
ok      github.com/ethereum/go-ethereum/ethclient       0.015s [no tests to run]
ok      github.com/ethereum/go-ethereum/ethdb   (cached)
?       github.com/ethereum/go-ethereum/ethstats        [no test files]
ok      github.com/ethereum/go-ethereum/event   (cached)
ok      github.com/ethereum/go-ethereum/event/filter    (cached)
?       github.com/ethereum/go-ethereum/internal/build  [no test files]
?       github.com/ethereum/go-ethereum/internal/cmdtest        [no test files]
?       github.com/ethereum/go-ethereum/internal/debug  [no test files]
?       github.com/ethereum/go-ethereum/internal/ethapi [no test files]
ok      github.com/ethereum/go-ethereum/internal/guide  6.683s
ok      github.com/ethereum/go-ethereum/internal/jsre   (cached)
?       github.com/ethereum/go-ethereum/internal/jsre/deps      [no test files]
?       github.com/ethereum/go-ethereum/internal/web3ext        [no test files]
ok      github.com/ethereum/go-ethereum/les     78.020s
?       github.com/ethereum/go-ethereum/les/flowcontrol [no test files]
ok      github.com/ethereum/go-ethereum/light   2.622s
?       github.com/ethereum/go-ethereum/log     [no test files]
?       github.com/ethereum/go-ethereum/log/term        [no test files]
ok      github.com/ethereum/go-ethereum/metrics (cached)
?       github.com/ethereum/go-ethereum/metrics/exp     [no test files]
?       github.com/ethereum/go-ethereum/metrics/influxdb        [no test files]
?       github.com/ethereum/go-ethereum/metrics/librato [no test files]
ok      github.com/ethereum/go-ethereum/miner   0.015s
ok      github.com/ethereum/go-ethereum/mobile  0.022s
ok      github.com/ethereum/go-ethereum/node    0.628s
ok      github.com/ethereum/go-ethereum/p2p     0.150s
ok      github.com/ethereum/go-ethereum/p2p/discover    17.179s
ok      github.com/ethereum/go-ethereum/p2p/discv5      8.621s
ok      github.com/ethereum/go-ethereum/p2p/enr 0.017s
ok      github.com/ethereum/go-ethereum/p2p/nat (cached)
ok      github.com/ethereum/go-ethereum/p2p/netutil     (cached)
ok      github.com/ethereum/go-ethereum/p2p/protocols   0.057s
ok      github.com/ethereum/go-ethereum/p2p/simulations 0.214s
ok      github.com/ethereum/go-ethereum/p2p/simulations/adapters        0.025s
?       github.com/ethereum/go-ethereum/p2p/simulations/examples        [no test files]
?       github.com/ethereum/go-ethereum/p2p/simulations/pipes   [no test files]
?       github.com/ethereum/go-ethereum/p2p/testing     [no test files]
ok      github.com/ethereum/go-ethereum/params  (cached)
ok      github.com/ethereum/go-ethereum/rlp     (cached)
ok      github.com/ethereum/go-ethereum/rpc     (cached)
ok      github.com/ethereum/go-ethereum/signer/core     2.775s
ok      github.com/ethereum/go-ethereum/signer/rules    0.147s
?       github.com/ethereum/go-ethereum/signer/rules/deps       [no test files]
ok      github.com/ethereum/go-ethereum/signer/storage  (cached)
ok      github.com/ethereum/go-ethereum/swarm   75.146s
ok      github.com/ethereum/go-ethereum/swarm/api       0.098s
ok      github.com/ethereum/go-ethereum/swarm/api/client        0.199s
ok      github.com/ethereum/go-ethereum/swarm/api/http  0.181s
ok      github.com/ethereum/go-ethereum/swarm/bmt       (cached)
ok      github.com/ethereum/go-ethereum/swarm/fuse      0.022s
?       github.com/ethereum/go-ethereum/swarm/log       [no test files]
?       github.com/ethereum/go-ethereum/swarm/metrics   [no test files]
ok      github.com/ethereum/go-ethereum/swarm/multihash (cached)
ok      github.com/ethereum/go-ethereum/swarm/network   3.538s
ok      github.com/ethereum/go-ethereum/swarm/network/bitvector (cached)
ok      github.com/ethereum/go-ethereum/swarm/network/priorityqueue     (cached)
ok      github.com/ethereum/go-ethereum/swarm/network/simulation        40.626s
ok      github.com/ethereum/go-ethereum/swarm/network/simulations       0.019s
ok      github.com/ethereum/go-ethereum/swarm/network/simulations/discovery     12.088s
ok      github.com/ethereum/go-ethereum/swarm/network/stream    26.292s
ok      github.com/ethereum/go-ethereum/swarm/network/stream/intervals  (cached)
ok      github.com/ethereum/go-ethereum/swarm/pot       (cached)
ok      github.com/ethereum/go-ethereum/swarm/pss       25.846s
ok      github.com/ethereum/go-ethereum/swarm/pss/client        6.050s
ok      github.com/ethereum/go-ethereum/swarm/pss/notify        1.031s
?       github.com/ethereum/go-ethereum/swarm/services/swap     [no test files]
ok      github.com/ethereum/go-ethereum/swarm/services/swap/swap        (cached)
?       github.com/ethereum/go-ethereum/swarm/spancontext       [no test files]
ok      github.com/ethereum/go-ethereum/swarm/state     (cached)
ok      github.com/ethereum/go-ethereum/swarm/storage   25.533s
ok      github.com/ethereum/go-ethereum/swarm/storage/encryption        (cached)
?       github.com/ethereum/go-ethereum/swarm/storage/mock      [no test files]
ok      github.com/ethereum/go-ethereum/swarm/storage/mock/db   (cached)
ok      github.com/ethereum/go-ethereum/swarm/storage/mock/mem  (cached)
ok      github.com/ethereum/go-ethereum/swarm/storage/mock/rpc  (cached)
?       github.com/ethereum/go-ethereum/swarm/storage/mock/test [no test files]
ok      github.com/ethereum/go-ethereum/swarm/storage/mru       1.074s
?       github.com/ethereum/go-ethereum/swarm/testutil  [no test files]
?       github.com/ethereum/go-ethereum/swarm/tracing   [no test files]
?       github.com/ethereum/go-ethereum/swarm/version   [no test files]
ok      github.com/ethereum/go-ethereum/tests   0.016s
ok      github.com/ethereum/go-ethereum/trie    0.958s
ok      github.com/ethereum/go-ethereum/whisper/mailserver      0.192s
?       github.com/ethereum/go-ethereum/whisper/shhclient       [no test files]
ok      github.com/ethereum/go-ethereum/whisper/whisperv5       43.808s
ok      github.com/ethereum/go-ethereum/whisper/whisperv6       12.327s
```

File: node/defaults.go [hunk 1]
```diff
@@ -57,11 +57,11 @@ func DefaultDataDir() string {
...
 	if home != "" {
 		if runtime.GOOS == "darwin" {
-			return filepath.Join(home, "Library", "Ethereum")
+			return filepath.Join(home, "Library", "Callisto")
 		} else if runtime.GOOS == "windows" {
-			return filepath.Join(home, "AppData", "Roaming", "Ethereum")
+			return filepath.Join(home, "AppData", "Roaming", "Callisto")
 		} else {
-			return filepath.Join(home, ".ethereum")
+			return filepath.Join(home, ".callisto")
```
Comment (yograterol) [inline node/defaults.go:64]: will it work with multi-geth without issues?

File: consensus/ethash/consensus.go [hunk 2]
```diff
@@ -799,3 +848,35 @@ func GetBlockEra(blockNum, eraLength *big.Int) *big.Int {
...
+		r.Sub(r, header.Number)
+		r.Mul(r, blockReward)
+		r.Div(r, big8)
+		state.AddBalance(uncle.Coinbase, r)
+
+		r.Div(blockReward, big32)
+		reward.Add(reward, r)
+	}
+	// Activate Callisto hardfork
+	if config.IsCLOHF1(header.Number) {
```
Comment (yograterol) [inline consensus/ethash/consensus.go:877]: we could set address in variables and after the `if` we could run the function `.AddBalance`

And the treasury and stake reward will change too after HF1

Total treasury fee at this stage is 30% which includes:
● Development funding: 10%
● Cold staking allocation: 20%

Comment (ghost) [reply consensus/ethash/consensus.go:877]: @yograterol That's correct

File: node/defaults.go (hunk 1 above)
Comment (ghost) [reply node/defaults.go:64]: @yograterol Nope if you are going to migrate from CLO/1.1 to multi-geth you need to move default filepath as I've said already.

Comment (yograterol) [reply node/defaults.go:64]: Ok, we must write it in some wiki or doc.

File: consensus/ethash/consensus.go (hunk 2 above)
Comment (yograterol) [reply consensus/ethash/consensus.go:877]: @akx20000a ok, but where do you calc the new cloTreasury and cloStake amount?

**Comment by ghost** [comment]
@yograterol Reward changed for HF1

**Pull request merged**